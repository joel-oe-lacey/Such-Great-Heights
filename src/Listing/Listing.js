import React from 'react';
import './Listing.scss'

const Listing = ({ id, name, viewListingInfo }) => {
    return (
        <section className='listing'>
            <h3>{name}</h3>
            <img src={`../../images/${id}_a.jpg`} alt={name}/>
            <button onClick={() => viewListingInfo(id)}>View Details</button>
        </section>
    )
}

export default Listing;
