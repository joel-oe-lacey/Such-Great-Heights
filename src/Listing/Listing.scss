.listing {
    font-family: 'Work Sans', sans-serif;
    width: 25%;
    height: 50%;
    border: 2px solid #555B6E;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;
    background-color: #FFD6BA;
    border-radius: 5%;

    h3 {
        text-align: center;
        font-size: 1em;
        font-weight: bold;
    }

    img {
        width: 75%;
        height: 50%;
        border-radius: 3%;
    }

    button {
        width: 75%;
        height: 10%;
        margin-top: 10%;
        grid-area: button;
        // width: 35%;
        // height: 35%;
        border-radius: 4px;
        font-family: 'Work Sans', sans-serif;
        font-weight: 700;
        font-size: 1em;
        margin-right: 2%;
        background-color: #FAF9F9;
        cursor: pointer;
    }
}
