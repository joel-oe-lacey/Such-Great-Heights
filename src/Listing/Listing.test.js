import React from 'react';
import { render } from '@testing-library/react';
import Listing from './Listing.js';
import { shallow } from 'enzyme';

describe('Listing', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( <Listing /> );
  });

  it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })

})
