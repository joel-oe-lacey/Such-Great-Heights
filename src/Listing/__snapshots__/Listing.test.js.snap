// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing should match the snapshot 1`] = `
<section
  className="listing"
>
  <h3 />
  <img
    src="../../images/undefined_a.jpg"
  />
  <NavLink
    to="/areas/undefined/listings/undefined"
  >
    <button
      className="view-details-btn"
    >
      View Details
    </button>
  </NavLink>
</section>
`;
